### DISCUSSION TITLE
How can I get or calculate the gradient of the surface normal？

### DISCUSSION THREAD

**Discussion by Always-kimi** [idaholab/moose#27767] [Q&A Modules: Navier-Stokes] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A Navier Stokes is the most appropriate category for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (avoid screenshots if possible, triple back quotes before/after pasted text, etc)

### Question

Hi there,

Does anyone know how to calculate the gradient of the surface normal? In my application, I consider the surface tension force and it need to calculate the surface curvature. According the equation it is equal to the gradient of the surface normal.
However, I browsed the framework source file only found the `_normals` and `_ad_normals`.

**Comment by GiudGiud** [comment]
@lindsayad did you have a surface tension term in the melt pool?

**Reply by lindsayad** [reply, marked as answer]
Yes, but it's only computed through a correlation with the temperature. @Always-kimi we do compute curvatures in libMesh and in MOOSE assembly. You can retrieve curvatures with automatic differentiation (AD) information, via `_assembly.adCurvatures()`. We do also have an `Assembly` member for curvatures without AD, but we strangely don't have an API for it. It could be easily added

**Reply by Always-kimi** [reply]
Thank you very much! I found it.
